pkg/infinispan/cluster.go

```diff
@@ -101,13 +99,18 @@ func (c Cluster) GracefulShutdown(podName string) error {
 }
 
 // GetClusterMembers get the cluster members as seen by a given pod
-func (c Cluster) GetClusterMembers(podName string) ([]string, error) {
+func (c Cluster) GetClusterMembers(podName string) (members []string, err error) {
 	rsp, err, reason := c.Client.Get(podName, consts.ServerHTTPHealthPath, nil)
-	if err := validateResponse(rsp, reason, err, "getting cluster members", http.StatusOK); err != nil {
-		return nil, err
+	if err = validateResponse(rsp, reason, err, "getting cluster members", http.StatusOK); err != nil {
+		return
 	}
 
-	defer rsp.Body.Close()
+	defer func() {
+		if cerr := rsp.Body.Close(); cerr != nil {
+			err = cerr
+		}
```

Comment:
I don't think this is right as it means that the original `err` is ignored even if it's `!= nil`. Shouldn't it be:

```go
defer func() {
        ccerr := rsp.Body.Close()
        if err == nil {
            err = cerr
        }
}
```

Reply:
Oh, yeah, seems to look at the not related code. Will fix